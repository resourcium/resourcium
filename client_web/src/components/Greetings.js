import React from 'react';
import './Greetings.scss';
import { ReactComponent as Logo } from 'src/images/greetingsimage.svg';

import useSettings from 'src/hooks/useSettings';


export default function Greetings(props) {
    const settings = useSettings();
    return (
        <div className="Greetings">
            <Logo className="wiggle" style={{ '--greeting-bgd': settings? settings.squiggle: '#E67E22' }} />
            <p>
            Hi, <span className="name">{props.name}</span>
            </p>
        </div>
    );
}
