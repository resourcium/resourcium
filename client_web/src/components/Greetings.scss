.Greetings {
  height: 170px;

  .wiggle{
    fill: var(--greeting-bgd);
  }
  p {
    position: absolute;
    top: 0;
    font-size: 50px;
    padding-left: 10px;
    color: var(--greeting-text);
    margin: 30px 0 0 0;
  }

  .name {
    font-weight: 500;
  }

  @media (max-width: 425px) {
    p {
      font-size: 35px;
    }
  }
}

@media (max-width: 425px) {
  .Greetings{
  height: 125px;

    .wiggle{
      width:250px;
      height:150px;
    }

    p {
      font-size: 30px;
    }
  }
}

@media (min-width: 1000px){
  .Greetings{
    height: 200px;

    .wiggle{
      width: 400px;
      height: 300px;
    }

    p {
      font-size: 60px;
    }
  }
}