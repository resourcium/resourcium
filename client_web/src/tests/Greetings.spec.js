import { create } from 'react-test-renderer';
import React from 'react';

import Greetings from 'src/components/Greetings';

import { wrapProvider } from 'src/tests/util';

test('greetings snapshot matches', () => {
    expect(create(wrapProvider(<Greetings />))).toMatchSnapshot();
});
