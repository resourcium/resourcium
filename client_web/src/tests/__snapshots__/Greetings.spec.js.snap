// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greetings snapshot matches 1`] = `
<div
  className="Greetings"
>
  <svg
    className="wiggle"
    style={
      Object {
        "--greeting-bgd": "#E67E22",
      }
    }
  >
    greetingsimage.svg
  </svg>
  <p>
    Hi, 
    <span
      className="name"
    />
  </p>
</div>
`;
